### PR TITLE
Add #favorite_opp #rival #head_to_head functionality

### PR DESCRIPTION
**Pull request by trpubz**

### CODE REVIEW

**Review by bkchilidawg — APPROVED** [review]
looks great, nice work. almost there

**Comment by kameronk92** [comment]
Looks like there are some merge conflicts, can you please take care of these, commit and push and we'll get it merged?

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -156,5 +156,17 @@ def team_info(team_id)
     # require 'byebug'; byebug
     teams_hash[:teams_info][team_id]
   end
+
+  def head_to_head(team_id)
```
Comment (kameronk92) [inline lib/stat_tracker.rb:160]: great job using the same helper method for these 3 methods

File: lib/stats.rb [hunk 2]
```diff
@@ -184,15 +185,51 @@ def teams_info
...
 
+  def win_pct_opp
+    win_pct_opp = Hash.new { |hash, key| hash[key] = {} }  # {team_id: {head_to_head: {opp: win_pct},
+    #                                                                   favorite_opponent: opp string,
+    #                                                                   rival: opp string}}
+
+    game_team_size = @game_teams_data.size
+    @game_teams_data.each_with_index do |team_game, idx|
+      # a team_game will have a :game_id match in another row; look in the index next or before
+      # need to check that index in range or will hit NoMethodError when calling [:game_id] key
```
Comment (kameronk92) [inline lib/stats.rb:205]: great job sorting through these

File: lib/stats.rb [hunk 3]
```diff
@@ -184,15 +185,51 @@ def teams_info
...
+      win_pct_opp[team_game[:team_id]][:head_to_head][opp] << team_game[:result]
+    end
+
+    win_pct_opp.each do |team_id, win_data|
+      win_data[:head_to_head].transform_values! do |results|  # => array of "WIN" || "LOSS"
+        (results.count("WIN") / results.size.to_f).round(2)
+      end
+
+      fav_opp_id = win_data[:head_to_head].max_by { |_, pct| pct }[0]
+      rival_id = win_data[:head_to_head].min_by { |_, pct| pct }[0]
```
Comment (kameronk92) [inline lib/stats.rb:225]: nicely done here.

File: spec/stat_tracker_spec.rb [hunk 4]
```diff
@@ -199,5 +199,23 @@
...
+
+  describe "#favorite_opponent" do
+    it "returns the team name for the highest win pct against" do
+      expect(@stat_tracker.favorite_opponent("6")).to eq "Houston Dynamo"
+    end
+  end
+  
+  describe "#rival" do
+    it "returns the team name for the lowest win pct against" do
+      expect(@stat_tracker.rival("3")).to eq "FC Dallas"
```
Comment (kameronk92) [inline spec/stat_tracker_spec.rb:217]: clean to see these all returning strings

File: lib/stat_tracker.rb (hunk 1 above)
Comment (trpubz) [reply lib/stat_tracker.rb:160]: 🙏

File: lib/stats.rb (hunk 2 above)
Comment (trpubz) [reply lib/stats.rb:315]: 🙏

File: lib/stats.rb (hunk 3 above)
Comment (trpubz) [reply lib/stats.rb:335]: 🙏

File: spec/stat_tracker_spec.rb (hunk 4 above)
Comment (trpubz) [reply spec/stat_tracker_spec.rb:217]: 🙏

**Pull request merged**